java/ql/src/experimental/Security/CWE/CWE-094/ScriptInjection.qhelp

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+
+<overview>
+<p>The JavaScript Engine API has been available since the release of Java 6, which allows
```

Comment:
Isn't "Java script-engine" maybe a little bit better than "Java Script Engine"?

Reply:
> Isn't "Java script-engine" maybe a little bit better than "Java Script Engine"?

Thanks @intrigus-lgtm for reviewing this PR. I never saw the term "Java script-engine" before. Normally it's called JavaScript engine or JavaScript scripting engine e.g. [Oracle Nashorn](https://www.oracle.com/technical-resources/articles/java/jf14-nashorn.html).

Could you show some links/sites that the word "Java script-engine" is used?

Reply:
https://docs.oracle.com/javase/7/docs/api/javax/script/ScriptEngine.html
This class is called `ScriptEngine`. It's purpose is to provide Java with scripting functionalities.
So when you refer to the API defined by `ScriptEngine` it should be either `Java ScriptEngine API` or `Java script-engine API`, because it is NOT an `JavaScript Engine API`.
At least that's how I understand it...

Reply:
I see. I will change it to <code>Java ScriptEngine API</code> for the API.

Reply:
Yes you are right, my point was indeed that the API is for scripting languages in general and not restricted to JavaScript (applies also to my other review comments).

- The [package documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.scripting/javax/script/package-summary.html) uses the terms "scripting API" and "Java Scripting Engines". The [module documentation](https://docs.oracle.com/en/java/javase/11/docs/api/java.scripting/module-summary.html) uses "Scripting API".
- [JSR 223](https://jcp.org/en/jsr/detail?id=223) calls it "Scripting for the Java Platform".
- The [guide](https://docs.oracle.com/javase/8/docs/technotes/guides/scripting/prog_guide/api.html) calls it "Java Scripting API".

Even though the JSR name might be the most 'correct' one, it is probably difficult to include it any sentence without breaking grammar or making it pretty verbose.

Personally I think "Java Scripting API" would be best since that is used by multiple JDK guides and is pretty descriptive.

Reply:
Thanks @Marcono1234 for the detailed explanation. Will do.

Reply:
Updated